llvm/include/llvm/Analysis/GenericDomTreeUpdater.h

```diff
@@ -146,7 +147,19 @@ class GenericDomTreeUpdater {
   /// 2. It is illegal to submit any update that has already been submitted,
   /// i.e., you are supposed not to insert an existent edge or delete a
   /// nonexistent edge.
-  void applyUpdates(ArrayRef<typename DomTreeT::UpdateType> Updates);
+  void applyUpdates(ArrayRef<UpdateT> Updates);
+
+  /// Apply updates that the critical edge (FromBB, ToBB) has been
+  /// split with NewBB.
+  ///
+  /// \note Do not use this method with regular edges.
+  ///
+  /// \note This kind updates are incompatible with generic updates,
+  /// call this method will submit all generic updates in lazy mode.
```

Comment:
```suggestion
  /// calling this method will submit all generic updates in lazy mode.
```

Reply:
Oops, the comment here is outdated now, all update kinds are supported in lazy mode.